src/components/EditAbilityScores.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import 'airbnb-js-shims' // to allow jest to understand Object.entries for parsing the character objects
+
+import { Form, FormGroup, Col, Button, ButtonGroup, ControlLabel } from 'react-bootstrap'
+
+export default class EditAbilityScores extends Component {
+  constructor (props) {
+    super(props)
+
+    this.renderAbilityScorePackages = this.renderAbilityScorePackages.bind(this)
+    this.pointsRemaining = this.pointsRemaining.bind(this)
+    this.increaseScore = this.increaseScore.bind(this)
+    this.decreaseScore = this.decreaseScore.bind(this)
+    this.resetScores = this.resetScores.bind(this)
+  }
+
+  pointsRemaining () {
+    let points = 27
+
+    Object.values(this.props.abilityScores).map((score) => {
+      if (score === 15) {
+        points -= 9
+      } else if (score === 14) {
+        points -= 7
+      } else {
+        points -= (score - 8)
+      }
+      return null
+    })
+
+    return points
+  }
+
+  checkChangeStatus (ability, type) {
+    if (type === 'increase' && this.props.abilityScores[ability] >= 15) {
+      return true
+    }
+
+    if (type === 'decrease' && this.props.abilityScores[ability] <= 8) {
+      return true
+    }
+  }
+
+  increaseScore (ability) {
+    let score = this.props.abilityScores[ability]
+    const cost = score === 14 ? 2 : ((score + 1) - 8) - (score - 8)
```

Comment:
This logic appears to be repeated above in line 23. Would be good to consolidate this in one place.